external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -225,6 +243,25 @@ private long doSeek(TopicPartition tp, OffsetAndMetadata committedOffset) {
         }
     }
 
+    /**
+     * Checks If {@link OffsetAndMetadata} was committed by an instance of {@link KafkaSpout} in this topology.
+     * This info is used to decide if {@link FirstPollOffsetStrategy} should be applied
+     *
+     * @param committedOffset {@link OffsetAndMetadata} info committed to Kafka
+     * @return true if this topology committed this {@link OffsetAndMetadata}, false otherwise
+     */
+    private boolean isOffsetCommittedByThisTopology(OffsetAndMetadata committedOffset) {
+        try {
+            final KafkaSpout.Info info = JSON_MAPPER.readValue(committedOffset.metadata(), KafkaSpout.Info.class);
+            return info.getTopologyId().equals(context.getStormId());
+        } catch (IOException e) {
+            LOG.trace("Failed to deserialize {}. Error likely occurred because the last commit " +
+                "for this topic-partition was done using an earlier version of Storm. " +
+                "Default behavior compatible with earlier version", committedOffset, e);
```

Comment:
Nit: Defaulting to behavior compatible with earlier version

Reply:
Done